src/Symfony/Component/DependencyInjection/Alias.php

```diff
@@ -22,7 +22,12 @@ class Alias
      */
     public function __construct($id, $public = true)
     {
-        $this->id = strtolower($id);
+        if (!is_string($id)) {
```

Comment:
That's what was said also about container->get, then had to explicitly at string casts in our own tests. So, yes. A BC layer is a friendly default policy.